macao.py

```diff
@@ -25,22 +26,37 @@ def get_num_of_players():
                 else:
                     return num_of_players
 
+        def put_first_card_from_pool_on_table(pool, table):
+            table.append(pool.pop(random.randint(0, len(pool) - 1)))
+
         num_of_players = get_num_of_players()
         for num in range(1, num_of_players + 1):
             self.players.append(Player(num))
         for player in self.players:
             player.draw_hand(self.deck)
+
+
+        # putting first card on table:
+        put_first_card_from_pool_on_table(self.deck.pool, self.deck.table)
+
         self._current_game(self.deck)
 
+    def put_card_from_user_to_table(player, card_index, table):
+        table.append(player.hand.pop(card_index - 1))
+
     def _current_game(self, deck):
         while not self.game_ended:
-            print(f"\nCard on table: ")
             for player in self.players:
                 print(f"\nPlayer #{player.players_id}\n")
+                print(f"\nCard on table:")
+                print(f"{self.deck.table[len(self.deck.table) - 1]}\n")
                 print("Cards on you hand:")
+                count = 1
                 for card in player.hand:
-                    print(repr(card))
-                users_selection = input("\nChoose a card to be put on the table: ")
+                    print(f"{count}. {repr(card)}")
+                    count += 1
```

Comment:
Let's try to avoid using counters

Reply:
Use function "enumerate()" which returns tuple (index, val)